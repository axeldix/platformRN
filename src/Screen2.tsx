import {type FC, memo} from 'react';

import {Platform, Pressable, Text, View} from 'react-native';

import {useNavigation} from './utils';

const Screen2: FC = () => {
  const {goBack} = useNavigation();

  const screenName = 'Screen2';

  return (
    <View
      style={{
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'rgba(67,241,181,0.3)',
      }}>
      <Text
        style={{
          fontSize: 50,
          fontWeight: 'bold',
          marginBottom: 20,
        }}>
        {screenName}
      </Text>
      {Platform.OS === 'web' && (
        <Pressable
          style={{
            backgroundColor: '#b8e378',
            padding: 10,
            borderRadius: 5,
          }}
          onPress={() => goBack()}>
          <Text
            style={{
              fontSize: 20,
              fontWeight: 'bold',
              color: 'white',
            }}>
            Go Back
          </Text>
        </Pressable>
      )}
    </View>
  );
};

export default memo(Screen2);
