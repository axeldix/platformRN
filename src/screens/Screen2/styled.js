import styled from 'styled-components/native';
import {Pressable, Text} from 'react-native';

export const Wrapper = styled.View`
  flex: 1;
  align-items: center;
  justify-content: center;
  background-color: rgba(67,241,181,0.3);
`;

export const Title = styled.Text`
  font-size: 50px;
  font-weight: bold;
  margin-bottom: 20px;
`;

export const StyledButton = styled(Pressable)`
  background-color: #b8e378;
  padding: 10px;
  border-radius: 5px;
`;

export const StyledButtonText = styled(Text)`
  font-size: 20px;
  font-weight: bold;
  color: white;
`;