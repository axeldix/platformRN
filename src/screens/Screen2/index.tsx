import {type FC, memo} from 'react';

import {Platform} from 'react-native';
import {StyledButton, StyledButtonText, Title, Wrapper} from './styled';

import {useNavigation} from '../../utils';

const Screen2: FC = () => {
  const {goBack} = useNavigation();

  const screenName = 'Screen2';

  return (
    <Wrapper>
      <Title>{screenName}</Title>
      {Platform.OS === 'web' && (
        <StyledButton onPress={() => goBack()}>
          <StyledButtonText>Go Back</StyledButtonText>
        </StyledButton>
      )}
    </Wrapper>
  );
};

export default memo(Screen2);
