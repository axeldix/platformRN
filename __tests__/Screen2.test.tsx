import React from 'react';
import Screen2 from '../src/screens/Screen2';
import {describe, it, expect} from '@jest/globals';
import renderer from 'react-test-renderer';

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(() => ({})),
}));
describe('Screen2', () => {
  it('should render the same as the snapshot', () => {
    const tree = renderer.create(<Screen2 />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
