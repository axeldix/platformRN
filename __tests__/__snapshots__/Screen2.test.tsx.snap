// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen2 should render the same as the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(67,241,181,0.3)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 50,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Screen2
  </Text>
</View>
`;
